### PR TITLE
Kvstore feedback changes

### PR DESCRIPTION
**Pull request by suryakumar1024**
KvStore code restructure

### CODE REVIEW

File: py-utils/src/utils/kv_store/conf_cache.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class ConfCache:
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/conf_cache.py:1]: Please add copyright license notice to all the files.

File: py-utils/src/utils/kv_store/conf_cache.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+class ConfCache:
+    ''' implements a ConfCache in specified format. '''
+
+    def __init__(self, doc):
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/conf_cache.py:4]: def __init__(self, kvstore):

File: py-utils/src/utils/kv_store/conf_cache.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        if k[0] not in data.keys() or type(data[k[0]]) != self._doc._type:
+            data[k[0]] = {}
+        self._set(k[1], val, data[k[0]])
+
+    def set(self, key, val):
+        ''' Sets the value into the DB for the given key '''
+        self._set(key, val, self._data)
+        self._dirty = True
+
+    def convert(self, map, payload):
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/conf_cache.py:45]: This interface is not required. ConfCache itself is payload.

File: py-utils/src/utils/kv_store/file_storage/file_kv_storage.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import os, errno, sys
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_storage/file_kv_storage.py:1]: Please keep all the files in kv_store directory itself. No need to create file_storage.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/file_storage/file_kv_storage.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import os, errno, sys
+
+class FileKvStorage:
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_storage/file_kv_storage.py:3]: Please add comment to all the classes.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/kv_store.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
+from src.utils.kv_store.kv_storage import KvStorage
+
+class KvStore:
+    """
+    This class will take KvStorage implementation as input and be front facing
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_store.py:6]: Need to be accurate and precise

"Class for Abstraction over all kinds of KV based Storage"

File: py-utils/src/utils/kv_store/file_storage/yaml_kv_storage.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+from src.utils.kv_store.file_storage.file_kv_storage import FileKvStorage
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_storage/yaml_kv_storage.py:1]: No need to separate classes into separate files. You should merge similar category classes into one file.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/conf_cache.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/conf_cache.py:18]: Updated with copyright license notice

File: py-utils/src/utils/kv_store/conf_cache.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/conf_cache.py]: Updated doc to kvstore

File: py-utils/src/utils/kv_store/conf_cache.py (hunk 3 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/conf_cache.py]: Removed convert API from the confcache

File: py-utils/src/utils/kv_store/file_storage/file_kv_storage.py (hunk 4 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/file_storage/file_kv_storage.py]: Updated directory structure as metioned

File: py-utils/src/utils/kv_store/kv_store.py (hunk 6 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/kv_store.py]: noted. will updated that

File: py-utils/src/utils/kv_store/file_storage/yaml_kv_storage.py (hunk 7 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/file_storage/yaml_kv_storage.py]: merged into single file

File: py-utils/src/utils/kv_store/file_storage/file_kv_storage.py (hunk 5 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/file_storage/file_kv_storage.py]: Added comments to all pre-implemented classes

File: py-utils/src/utils/kv_store/file_storage/json_kv_storage.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
+from src.utils.kv_store.file_storage.file_kv_storage import FileKvStorage
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_storage/json_kv_storage.py]: Merge this into the file "kv_store/file_kv_storage.py"

File: py-utils/src/utils/kv_store/file_storage/file_kv_storage_factory.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_storage/file_kv_storage_factory.py]: Please merge this with kv_store/file_kv_storage.py

File: py-utils/src/utils/kv_store/conf_cache.py (hunk 1 above)
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/conf_cache.py:18]: This file should be part of conf_store module and not kv_store

File: py-utils/src/utils/kv_store/kv_store.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        self._storage = kvStorage
+
+    def get(self, key):
+        return self._storage.get(key)
+
+    def set(self, key, value):
+        return self._storage.set(key, value)
+
+    def delete(self, key):
+        return self._storage.delete(key)
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_store.py:35]: Please add test code.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/file_kv_storage_factory.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/file_kv_storage_factory.py:1]: Merge this into file_kv_storage.py. Refer to code example, I provided to know how to do this.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/kv_storage_collections.py [hunk 12]
```diff
@@ -0,0 +1,129 @@
+from src.utils.kv_store.file_kv_storage import FileKvStorage
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_storage_collections.py:1]: Rename this file to kv_file_storage_collections.py. There will be another collection for dist_kv_storage.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/kv_storage_collections.py [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+    def dump(self, data: dict):
+        """
+        Set's the data _source after converting to json
+        :param data: :type: Dict
+        :return:
+        """
+        self._file_path = json.dumps(data)
+
+class TextKvStorage(FileKvStorage):
+
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_storage_collections.py:116]: Remove blank line. Please be consistent w.r.t. spacing.

File: py-utils/src/utils/kv_store/kv_storage_collections.py [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+    def load(self):
+        return self._file_path
+
+    def dump(self, data):
+        self._file_path = data
+
+class JsonMessageKvStorage(Json):
+    def __init__(self, json_str):
+        """
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_storage_collections.py:94]: Please use """ or '''. Please be consistent

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Comment by suryakumar1024** [comment]
Closing this PR since change in architecture

**Pull request closed without merging**